test/unit/nedap/utils/spec/api/check.cljc

```diff
@@ -0,0 +1,50 @@
+(ns unit.nedap.utils.spec.api.check
+  (:require
+   #?(:clj [clojure.spec.alpha :as spec] :cljs [cljs.spec.alpha :as spec])
+   #?(:clj [clojure.test :refer [deftest testing are is use-fixtures]] :cljs [cljs.test :refer-macros [deftest testing is are] :refer [use-fixtures]])
+   [nedap.utils.spec.api :as sut])
+  #?(:clj (:import (clojure.lang ExceptionInfo))))
+
+(spec/def ::number number?)
+
+(defn backed-by-dynamic-spec-checking
+  "A function that performs an `assert` using a dynamic spec.
+
+  This challenges our `check!` helper: it shouldn't report the `'spec` symbol as the relevant spec,
+  but the underlying spec value instead."
+  [spec x]
+  (assert (sut/check! spec x))
+  x)
+
+(def sample-spec-object (spec/spec #{1 2 3}))
+
+(deftest reported-ex-data
+
+  (are [passed-spec expected] (testing [passed-spec expected]
+                                (try
+                                  (with-out-str
+                                    (backed-by-dynamic-spec-checking passed-spec "NaN"))
+                                  (is false)
+                                  (catch ExceptionInfo e
+                                    (let [actual (-> e ex-data (select-keys [:spec-object
+                                                                             :quoted-spec
+                                                                             :faulty-value-object
+                                                                             :quoted-faulty-value]))]
+                                      (is (= expected actual)))))
```

Comment:
use `(is (match? expected actual))`. It's more extensible. and prevents manually selecting keys

Reply:
I'm open to `match?` but I haven't gotten to study it yet (other than a reasonable surface understanding of how it works, obviously).

Fortunately `match?` is less fuzzy/ambiguous than I initially thought, but I still have hesitation around whether it possibly could give us false positives/negatives someday. Particularly as time passes; IME often not sufficient attention is paid to how tests are understood or changed across the many months/years of their lifetime.

One thing that I definitely like of `select-keys` is that the resulting test is definitely unambiguous; exact matches make unmistakable tests (comparing _values_ is stronger than comparing _computations_). Typing `select-keys` seems a very low cost for obtaining that.

I'd invite you and @lennartbuit to open a PR against https://github.com/nedap/clojure-guidelines/ explaining why and when to use it, how it compares to select-keys, etc. Most definitely I'd 👍it, my only hesitation is how universal its use case is.